src/particle/particle/particle.py

```diff
@@ -316,53 +312,68 @@ def dump_table(
+            which can be any `Particle` class property.
         exclude_fields: list, optional, default is []
             List of table fields to exclude in the printout.
             Relevant only when exclusive_fields is not given.
         n_rows: int, optional, defaults to all rows
             Number of table rows to print out.
         filter_fn: function, optional, default is None
             Apply a filter to each particle.
-            See findall(...) for typical use cases.
-        filename: str, optional, default is None
-            Name of file where to dump the table.
-            By default the table is dumped to stdout.
-        tablefmt: str, optional, default is 'simple'
-            Table formatting option, see the tabulate's package
-            tabulate function for a description of available options.
-            The most common options are:
-            'plain', 'simple', 'grid', 'rst', 'html', 'latex'.
-        floatfmt: str, optional, default is '.12g'
-            Number formatting, see the tabulate's package
-            tabulate function for a description of available options.
-        numalign: str or None, oprional, default is 'decimal'
-            Column alignment for numbers, see the tabulate's package
-            tabulate function for a description of available options.
+            See `findall(...)`` for typical use cases.
 
         Returns
         -------
-        str or None if filename is None or not, respectively.
+        The particle table query as a `list`.
 
         Note
         ----
-        Uses the `tabulate` package.
+        The `tabulate` package is suggested as a means to print-out
+        the contents of the query as a nicely formatted table.
 
         Examples
         --------
-        print(Particle.dump_table())
-        print(Particle.dump_table(n_rows=5))
-        print(Particle.dump_table(exclusive_fields=['pdgid', 'pdg_name']))
-        print(Particle.dump_table(filter_fn=lambda p: p.pdgid.has_bottom))
-        Particle.dump_table(filename='output.txt', tablefmt='rst')
-        """
-        from tabulate import tabulate
+        Reproduce the whole particle table kept internally:
+
+        >>> Particle.to_list()    # doctest: +SKIP
+
+        Reduce the information on the particle table to the only fields
+        ['pdgid', 'pdg_name'] and render the first 5 particles:
+
+        >>> query_as_list = Particle.to_list(exclusive_fields=['pdgid', 'pdg_name'], n_rows=5)
+        >>> from tabulate import tabulate    # doctest: +SKIP
+        >>> print(tabulate(query_as_list, headers='firstrow'))    # doctest: +SKIP
+
+        Request the properties of a specific list of particles:
+
+        >>> query_as_list = Particle.to_list(filter_fn=lambda p: p.pdgid.is_lepton and p.charge!=0, exclusive_fields=['pdgid', 'name', 'mass', 'charge'])
+
+        >>> print(tabulate(query_as_list, headers='firstrow', tablefmt="rst", floatfmt=".12g", numalign="decimal"))    # doctest: +SKIP
```

Comment:
You can add tabulate as a doc requirement, then not skip this.

Reply:
OK, done.